app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,230 @@
-            binding.messageButton.setOnClickListener {
-                interactor.onMessageClicked(message)
+        MessageCardPlaceHolder(
+            message = messageGlobal,
+            onMessageClicked = { interactor.onMessageClicked(messageGlobal) },
+            onMessageClosedClicked = { interactor.onMessageClosedClicked(messageGlobal) }
+        )
+    }
+}
+
+/**
+ * Message Card.
+ *
+ * @param message contains title and description of message.
+ * @param onMessageClicked Invoked when user clicks on the button.
+ * @param onMessageClosedClicked Invoked when user clicks on close button to remove message.
+ */
+@Suppress("LongMethod")
+@Composable
+fun MessageCardPlaceHolder(
+    message: Message,
+    onMessageClicked: () -> Unit,
+    onMessageClosedClicked: () -> Unit,
+) {
+    Card(
+        modifier = Modifier
+            .padding(16.dp)
+            .then(
+                if (message.data.buttonLabel.isNullOrBlank()) {
+                    Modifier.clickable(onClick = onMessageClicked)
+                } else {
+                    Modifier
+                }
+            ),
+        shape = RoundedCornerShape(8.dp),
+        backgroundColor = FirefoxTheme.colors.layer2,
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            if (!message.data.title.isNullOrBlank()) {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+
+                    SectionHeader(
+                        text = message.data.title!!,
+                        modifier = Modifier
+                            .weight(1f)
+                            .padding(bottom = 10.dp)
+                    )
+
+                    IconButton(
+                        modifier = Modifier.size(20.dp),
+                        onClick = onMessageClosedClicked
+                    ) {
+                        Icon(
+                            painter = painterResource(R.drawable.ic_close),
+                            contentDescription = stringResource(
+                                R.string.remove_home_collection_placeholder_content_description
+                            ),
+                            tint = FirefoxTheme.colors.iconPrimary
+                        )
+                    }
+                }
+                PrimaryText(
+                    text = message.data.text,
+                    modifier = Modifier.fillMaxWidth(),
+                    fontSize = 14.sp
+                )
+            } else {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+                    PrimaryText(
+                        text = message.data.text,
+                        modifier = Modifier.weight(1f),
+                        fontSize = 14.sp
+                    )
+                    IconButton(
+                        modifier = Modifier.size(24.dp),
```

Comment:
drive by: I noticed that the IconButton here is 24.dp but the one in the positive `if` is 20.dp. Not sure if that's intentional

Reply:
I was changing it to 20dp in the next push :P

Reply:
Good catch! Thank you